### PR TITLE
HAC-1617: Workspace details

### PR DESCRIPTION
**Pull request by kdoberst**
Adds a page to view workspace details.  

**Video:**

https://user-images.githubusercontent.com/82059948/197555624-43c1bb4f-4f0d-4c2e-b659-1e845ed25193.mov

NOTE:  This PR assumes that you are using the local proxy. Also, this cannot be merged until the following PRs are merged and published:  https://github.com/openshift/dynamic-plugin-sdk/pull/168 and https://github.com/openshift/dynamic-plugin-sdk/pull/175.  This PR will fail tests until these PRs are resolved.

### CODE REVIEW

File: src/components/WorkspaceDetails/WorkspaceDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        </OverviewPage>
+      ),
+      ariaLabel: 'Overview',
+    },
+  ];
+};
+
+const WorkspaceDetailsBreadcrumbs = (workspaceName: string) => [
+  { name: 'Workspaces', path: '/workspaces' },
+  { name: 'Workspace details', path: workspaceName ? `/workspaces/${workspaceName}` : '' },
```
Comment (rhamilto) [inline src/components/WorkspaceDetails/WorkspaceDetails.tsx:29]:
```suggestion
  { name: 'Workspace Details', path: workspaceName ? `/workspaces/${workspaceName}` : '' },
```
if https://www.patternfly.org/v4/ux-writing/capitalization#capitalization-across-red-hat-uis is what applies here.  Is it just me, or do these rules keep changing?

File: src/pages/WorkspaceDetails/WorkspaceDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { Helmet } from 'react-helmet';
+import { PageSection, PageSectionVariants } from '@patternfly/react-core';
+import { WorkspaceDetails } from '../../components/WorkspaceDetails';
+
+const Workspaces: React.FC = () => {
+  const title = 'Workspace Details';
+
+  return (
+    <>
+      <Helmet>
```
Comment (rhamilto) [inline src/pages/WorkspaceDetails/WorkspaceDetails.tsx:11]: Move this to `WorkspaceDetails` so we can include the workspace name (which will be helpful for bookmarking, etc)?

File: src/components/WorkspaceDetails/WorkspaceDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+      })
+      .finally(() => {
+        setLoaded(true);
+      });
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <div style={{ overflow: 'scroll' }}>
```
Comment (rhamilto) [inline src/components/WorkspaceDetails/WorkspaceDetails.tsx:90]: External style since we should avoid inline styles and we're likely to need this in other pages?

File: src/components/WorkspaceDetails/WorkspaceDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+import type { K8sResourceCommon } from '@openshift/dynamic-plugin-sdk-utils';
+import { useParams } from 'react-router-dom';
+
+const WorkspaceDetailsTabs = (workspace: K8sResourceCommon, loaded: boolean, error: { message: string; status: number }) => {
+  return [
+    {
+      key: 'overview',
+      title: 'Overview',
+      content: (
+        <OverviewPage loaded={loaded} resource={workspace} loadError={error} rightColumn={<></>}>
```
Comment (rhamilto) [inline src/components/WorkspaceDetails/WorkspaceDetails.tsx:18]: Should use `OverviewTab`?  It looks like `OverviewTab` is only being used in tests.

**Review by rhamilto — COMMENTED** [review]
Nice job, @kdoberst!  I gave this a cursory review and will go in greater depth once you have the necessary dependency update included.

File: src/components/WorkspaceAdd/WorkspaceAddButton.tsx [hunk 5]
```diff
@@ -62,11 +65,12 @@ const WorkspaceAddButton = ({ workspaces }: { workspaces: K8sResourceCommon[] })
     })
       .then((response) => {
         // eslint-disable-next-line no-console
-        console.log(
-          // @ts-ignore
-          `You have created a ${response.spec.type.name} workspace with a name of ${response.metadata.name} and an UUID of ${response.metadata.uid}`,
-        );
+        // console.log(
```
[the remark this answers is not in the document]
Comment (kdoberst) [reply src/components/WorkspaceAdd/WorkspaceAddButton.tsx:68]: Yep - good point.  I had put that in so the user would know something happened.  Now that the code goes to the details page that lets the user know the workspace was created.

File: src/components/WorkspaceDetails/WorkspaceDetails.tsx (hunk 1 above)
Comment (kdoberst) [reply src/components/WorkspaceDetails/WorkspaceDetails.tsx:29]: Good question.  There are multiple designs out there and see what they have.

File: src/components/WorkspaceDetails/WorkspaceDetails.tsx (hunk 3 above)
Comment (kdoberst) [reply src/components/WorkspaceDetails/WorkspaceDetails.tsx:90]: Ha - it is used on other pages, so I copied what was used there.  Instead of using an external CSS, how do we feel about incorporating styled-components?  A find it to be a much cleaner way to incorporate special case CSS.

File: src/components/WorkspaceDetails/WorkspaceDetails.tsx (hunk 4 above)
Comment (kdoberst) [reply src/components/WorkspaceDetails/WorkspaceDetails.tsx:18]: I'm trying to set up a pattern for future use.  In theory, there will be more than the overview tab. So if each tab has a separate file we prevent having a lot of tab-specific code on the WorkspaceDetails page.

File: src/components/WorkspaceDetails/WorkspaceDetails.tsx (hunk 1 above)
Comment (kdoberst) [reply src/components/WorkspaceDetails/WorkspaceDetails.tsx:29]: It is all over the place - so let's just change it to lower case to match the patternfly doc.

File: src/pages/WorkspaceDetails/WorkspaceDetails.tsx (hunk 2 above)
Comment (kdoberst) [reply src/pages/WorkspaceDetails/WorkspaceDetails.tsx:11]: So I'm torn over what should be the best pattern moving forward.  Looks like files in the `pages` directory determine the overall wrapper/layout for the page.  

Wondering if instead of putting the code including `helmet` in the component, we pull the workspace name from useParams.

Comment (rhamilto) [reply src/pages/WorkspaceDetails/WorkspaceDetails.tsx:11]: I don't have a strong opinion on either approach (`<Helmet>` is designed to allow lower level components to override, but that doesn't mean it's necessarily a good pattern), but do think we should include the workspace name in `<title>`.

File: src/components/WorkspaceDetails/WorkspaceDetails.tsx (hunk 3 above)
Comment (rhamilto) [reply src/components/WorkspaceDetails/WorkspaceDetails.tsx:90]: A styled component makes sense for reuse, but I still think the associated styles should go in a separate scss file.

Comment (kdoberst) [reply src/components/WorkspaceDetails/WorkspaceDetails.tsx]: I would say take a look at what I changed.  It is easier to tie specific css to a particular component so you have control when the css is applied.  Secondly styled components only load the css the first time it is used.  So it isn't called if the components is never called and it won't call the css more than once if has already been used.

File: src/components/WorkspaceDetails/WorkspaceDetails.tsx (hunk 4 above)
Comment (kdoberst) [reply src/components/WorkspaceDetails/WorkspaceDetails.tsx]: After we talked - I see that you are pointing out what I intended to do, but didn't actually connect the files.  The correct imports have been included.

File: package.json [hunk 6]
```diff
@@ -48,23 +51,22 @@
     "jest": "^27.4.2",
     "jest-axe": "^6.0.0",
     "npm-run-all": "4.1.5",
+    "react-redux": "^7.2.2",
```
Comment (kdoberst) [inline package.json:54]: Just noticed this because it moved.  Do we really need redux?  I don't see anywhere it is used.

File: src/components/StyledComponents/PageContentWrapper.tsx [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import styled from 'styled-components';
+
+const PageContentWrapper = styled.div`
```
Comment (rhamilto) [inline src/components/StyledComponents/PageContentWrapper.tsx:3]: My initial reaction to this is yuck!  Why?  I worked for Steve Champeon, who coined the phrase "progressive enhancement", so this seems like a fundamental separation of concerns issue.  That said, I do realize React is significantly blurring the separation of concerns lines, but it seems like this is going all in on it.

Additionally, it seems like hac-dev and dynamic-plugin-sdk-utils are using external styles, so I think it's better to use one approach for consistency.

Comment (kdoberst) [reply src/components/StyledComponents/PageContentWrapper.tsx:3]: The ES 6 syntax can be jarring if you aren't used to it.  There is a way to pass a JSON of styles, but you don't see that used too much because who really likes the scss code in json format?

Here is where I think we should move forward.  I think the use of styled components (or something similar) would be extremely helpful.  It might be obvious in my simple example but actually leads to a better separation of structure and styling.  I'll revert this to an external CSS file, but I can present about styled-components in an upcoming meeting so the benefits can be more fully shown and discussed.

File: package.json [hunk 8]
```diff
@@ -23,22 +23,25 @@
     "@patternfly/react-core": "^4.192.15",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
-    "react-helmet": "^6.1.0"
+    "react-helmet": "^6.1.0",
+    "styled-components": "^5.3.6"
```
Comment (florkbr) [inline package.json:27]: TODO remove style-components and deliver in a follow up PR for discussions

File: src/components/StyledComponents/PageContentWrapper.tsx (hunk 7 above)
Comment (florkbr) [reply src/components/StyledComponents/PageContentWrapper.tsx:3]: I think is a great discussion - we should probably move this to a separate PR and have a discussion with a broader team. I personally have used https://cssinjs.org/?v=v10.9.2 on previous teams - and it worked wonders to prevent styles from bleeding outside of components un-intentionally.

File: src/components/WorkspaceAdd/WorkspaceAddButton.test.tsx [hunk 9]
```diff
@@ -286,20 +300,25 @@ describe('Add Workspace modal', () => {
         },
       },
     });
-    await waitFor(() => expect(screen.queryByRole('dialog')).not.toBeInTheDocument());
+
+    await waitFor(() => expect(mockedUsedNavigate).toHaveBeenCalledWith('1workspace'));
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+    // KKD - verify route change
```
Comment (florkbr) [inline src/components/WorkspaceAdd/WorkspaceAddButton.test.tsx:306]: @kdoberst can we remove this if we are verifying the URL?

File: src/components/WorkspaceAdd/WorkspaceAddButton.tsx [hunk 10]
```diff
@@ -61,12 +64,8 @@ const WorkspaceAddButton = ({ workspaces }: { workspaces: K8sResourceCommon[] })
       },
     })
       .then((response) => {
-        // eslint-disable-next-line no-console
-        console.log(
-          // @ts-ignore
-          `You have created a ${response.spec.type.name} workspace with a name of ${response.metadata.name} and an UUID of ${response.metadata.uid}`,
-        );
         setIsModalOpen(!isModalOpen);
+        navigate(`${response.metadata.name}`);
```
[the remark this answers is not in the document]
Comment (florkbr) [reply src/components/WorkspaceAdd/WorkspaceAddButton.tsx:68]: Open Q - we should ask KCP if all workspace names will be URL safe in the long term.

File: src/components/WorkspaceDetails/OverviewTab.tsx [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+type OverviewTabProps = {
+  loaded: boolean;
+  resource: K8sResourceCommon;
+  error: { message: string; status: number };
+};
+
+const OverviewTab = ({ loaded, resource, error }: OverviewTabProps) => {
+  return (
+    <OverviewPage loaded={loaded} resource={resource} loadError={error} rightColumn={<></>}>
+      <DetailsItem label="Services" resource={resource} path="spec.type.name" loaded={loaded} />
```
Comment (florkbr) [inline src/components/WorkspaceDetails/OverviewTab.tsx:13]: @kdoberst can we put this in a constant?

File: src/components/WorkspaceDetails/OverviewTab.tsx [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { OverviewPage, DetailsItem, K8sResourceCommon } from '@openshift/dynamic-plugin-sdk-utils';
+
+type OverviewTabProps = {
+  loaded: boolean;
+  resource: K8sResourceCommon;
+  error: { message: string; status: number };
```
Comment (florkbr) [inline src/components/WorkspaceDetails/OverviewTab.tsx:7]: @kdoberst can you see if the SDK exposes a type for this? If not - we probably need to create an issue to expose a type for it on the Core teams backlog.

File: src/pages/WorkspaceDetails/WorkspaceDetails.tsx [hunk 13]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { Helmet } from 'react-helmet';
+import { PageSection, PageSectionVariants } from '@patternfly/react-core';
+import { WorkspaceDetails } from '../../components/WorkspaceDetails';
+import { useParams } from 'react-router-dom';
+
+const Workspaces: React.FC = () => {
+  const title = 'Workspace Details';
+  const { workspaceName } = useParams();
```
Comment (florkbr) [inline src/pages/WorkspaceDetails/WorkspaceDetails.tsx:9]: TODO for Bryan - chat with @vojtechszocs to see if there is a better way of passing props to the coderef.

File: src/pages/WorkspaceDetails/WorkspaceDetails.test.tsx [hunk 14]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { axe, toHaveNoViolations } from 'jest-axe';
+import WorkspaceDetailsPage from './WorkspaceDetails';
+expect.extend(toHaveNoViolations);
```
Comment (florkbr) [inline src/pages/WorkspaceDetails/WorkspaceDetails.test.tsx:6]: @kdoberst we can remove this since we put it in the JEST setup

File: src/pages/WorkspaceDetails/WorkspaceDetails.test.tsx [hunk 15]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
```
Comment (florkbr) [inline src/pages/WorkspaceDetails/WorkspaceDetails.test.tsx:3]: Also not needed

File: src/components/WorkspaceList/WorkspaceList.test.tsx [hunk 16]
```diff
@@ -1,11 +1,15 @@
...
-import { BrowserRouter } from 'react-router-dom';
+import { BrowserRouter, Router } from 'react-router-dom';
+import { createMemoryHistory } from 'history';
+import { toHaveNoViolations, axe } from 'jest-axe';
 import '@testing-library/jest-dom';
 import { WorkspaceList } from './index';
 import type { K8sResourceCommon } from '@openshift/dynamic-plugin-sdk-utils';
 import { useK8sWatchResource } from '@openshift/dynamic-plugin-sdk-utils';
 
+expect.extend(toHaveNoViolations);
```
Comment (florkbr) [inline src/components/WorkspaceList/WorkspaceList.test.tsx:11]: Can remove :)

File: src/components/WorkspaceList/WorkspaceList.test.tsx [hunk 17]
```diff
@@ -98,4 +102,44 @@ describe('Workspace list', () => {
...
+    useK8sWatchResourceMock.mockReturnValue([[...workspacesMockData], true, null]);
+
+    const { container } = render(
+      <BrowserRouter>
+        <WorkspaceList />
+      </BrowserRouter>,
+    );
+    expect(await screen.findByText('demo-ws1')).toBeInTheDocument();
+
+    // NOTE this test will fail until https://github.com/openshift/dynamic-plugin-sdk/pull/175
```
Comment (florkbr) [inline src/components/WorkspaceList/WorkspaceList.test.tsx:116]: Note can be removed as well (this has been merged in!)

File: src/components/WorkspaceList/WorkspaceList.test.tsx [hunk 18]
```diff
@@ -98,4 +102,44 @@ describe('Workspace list', () => {
...
+    // is merged and published
+    const results = await axe(container);
+    expect(results).toHaveNoViolations();
+  });
+
+  test('clicking on workspace name goes to details route', async () => {
+    useK8sWatchResourceMock.mockReturnValue([[...workspacesMockData], true, null]);
+
+    const history = createMemoryHistory();
+    history.push = jest.fn();
```
Comment (florkbr) [inline src/components/WorkspaceList/WorkspaceList.test.tsx:126]: Can we switch this to use `navigate` mock instead so we mirror what we have in the component itself.

File: src/components/WorkspaceDetails/WorkspaceDetails.tsx [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+  //   groupVersionKind: {
+  //     group: 'tenancy.kcp.dev',
+  //     version: 'v1beta1',
+  //     kind: 'Workspace',
+  //   },
+  // };
+
+  // console.log('KKD - workspaceName', workspaceName);
+  // const [workspaceWatch, loadedWatch, errorWatch] = useK8sWatchResource(watchedResource, {
+  //   wsPrefix: `ws/kcp/${workspaceName}`,
```
Comment (florkbr) [inline src/components/WorkspaceDetails/WorkspaceDetails.tsx:56]: We need to log an issue to enable this once we finish investigating the `wsPath` and `pathPrefix` options issues with our hook. CC @karelhala

File: src/components/WorkspaceDetails/WorkspaceDetails.tsx [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <PageContentWrapper>
+      <DetailsPage
+        ariaLabel="Workspace details"
+        tabs={WorkspaceDetailsTabs(workspace, loaded, error)}
+        breadcrumbs={WorkspaceDetailsBreadcrumbs(workspace?.metadata?.name)}
+        pageHeading={{ title: workspace?.metadata.name || workspaceName || 'Workspace Details' }}
```
Comment (florkbr) [inline src/components/WorkspaceDetails/WorkspaceDetails.tsx:87]: Might want to make `metadata.name` optional chaining again

File: src/components/WorkspaceDetails/WorkspaceDetails.tsx [hunk 21]
```diff
@@ -0,0 +1,93 @@
...
+        setLoaded(true);
+      });
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <PageContentWrapper>
+      <DetailsPage
+        ariaLabel="Workspace details"
```
Comment (florkbr) [inline src/components/WorkspaceDetails/WorkspaceDetails.tsx:84]: Can we put `"Workspace Details"` in a constant and use it here and below in the `pageHeading` conditional?

File: src/components/WorkspaceDetails/WorkspaceDetails.test.tsx [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+import { render, screen, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { axe, toHaveNoViolations } from 'jest-axe';
+import { Router } from 'react-router-dom';
+import { createMemoryHistory } from 'history';
+import type { MemoryHistory } from 'history';
+import { k8sGetResource } from '@openshift/dynamic-plugin-sdk-utils';
+import { WorkspaceDetails } from './index';
+
+expect.extend(toHaveNoViolations);
```
Comment (florkbr) [inline src/components/WorkspaceDetails/WorkspaceDetails.test.tsx:11]: TODO remove

File: src/components/WorkspaceDetails/WorkspaceDetails.test.tsx [hunk 23]
```diff
@@ -0,0 +1,116 @@
...
+    it('Error is displayed', async () => {
+      const err = { message: 'Sample Error', json: { code: 123 } };
+      k8sGetResourceMock.mockRejectedValue(err);
+
+      render(
+        <Router location={history.location} navigator={history}>
+          <WorkspaceDetails />
+        </Router>,
+      );
+      expect(await screen.findByText('Sample Error')).toBeInTheDocument();
```
Comment (florkbr) [inline src/components/WorkspaceDetails/WorkspaceDetails.test.tsx:112]: +1 for testing the failure case

File: src/components/WorkspaceDetails/WorkspaceDetails.test.tsx [hunk 24]
```diff
@@ -0,0 +1,116 @@
...
+      k8sGetResourceMock.mockResolvedValue(workspaceData);
+      const { container } = render(
+        <Router location={history.location} navigator={history}>
+          <WorkspaceDetails />
+        </Router>,
+      );
+      // Ensure that the useEffect has fully finished
+      expect(await screen.findByText('universal')).toBeInTheDocument();
+
+      // NOTE this test will fail until https://github.com/openshift/dynamic-plugin-sdk/pull/175
```
Comment (florkbr) [inline src/components/WorkspaceDetails/WorkspaceDetails.test.tsx:97]: Need to remove this comment

File: src/components/WorkspaceDetails/WorkspaceDetails.test.tsx [hunk 25]
```diff
@@ -0,0 +1,116 @@
...
+  let history: MemoryHistory;
+
+  beforeEach(() => {
+    jest.resetModules();
+    k8sGetResourceMock.mockClear();
+    history = createMemoryHistory();
+    history.push('/workspaces/demo-workspace');
+  });
+
+  it('Call to get information is made', async () => {
```
Comment (florkbr) [inline src/components/WorkspaceDetails/WorkspaceDetails.test.tsx:51]: Nit - we should investigate adding a lint rule to make sure we stay consistent with our spec function calls (use either `it` or `test` for consistency).

File: src/pages/WorkspaceDetails/WorkspaceDetails.test.tsx (hunk 14 above)
Comment (kdoberst) [reply src/pages/WorkspaceDetails/WorkspaceDetails.test.tsx:6]: Actually, we didn't add this to the test setup.  That was the SDK repo.   No worries - I'll set it up in HAC Infra and add it as a separate commit.

File: src/components/WorkspaceList/WorkspaceList.test.tsx (hunk 18 above)
Comment (kdoberst) [reply src/components/WorkspaceList/WorkspaceList.test.tsx:126]: I'm not sure why, but changing this to just mock useNavigate isn't working.  Maybe because useNavigate is inside another component ???

File: src/components/WorkspaceDetails/OverviewTab.tsx (hunk 12 above)
Comment (kdoberst) [reply src/components/WorkspaceDetails/OverviewTab.tsx:7]: The error is LoadError within the SDK and it is not exported outside.

File: src/components/WorkspaceDetails/WorkspaceDetails.test.tsx (hunk 25 above)
Comment (kdoberst) [reply src/components/WorkspaceDetails/WorkspaceDetails.test.tsx:51]: Test wins.  I'll make it consistent in Hac Infra.

If you are curious here are the numbers:

HAC-Infra:  3 files use it(), 4 files use test()
SDK: 6 files use it(), 10 files use test()
HAC-Core: 0 files use it(), 2 files use test()

**Comment by kdoberst** [comment]
@florkbr @rhamilto Made changes to this PR.  There were a few things I couldn't change - for those things I've put in the comments.

Because this PR is getting large I've kept the PR comments as separate commit(s).  Let me know if we want to squash those commits before merge time.

**Comment by florkbr** [comment]
@kdoberst for anything we weren't able to deliver in our follow up commits (that we would like to change in the future), can we create and link a Jira issue so we don't lose track? Thanks!

File: src/components/WorkspaceAdd/WorkspaceAddButton.tsx [hunk 26]
```diff
@@ -65,7 +65,11 @@ const WorkspaceAddButton = ({ workspaces }: { workspaces: K8sResourceCommon[] })
     })
       .then((response) => {
         setIsModalOpen(!isModalOpen);
-        navigate(`${response.metadata.name}`);
+        if (response?.metadata?.name) {
+          navigate(`${response.metadata.name}`);
+        } else {
+          throw new Error('Workspace successfully added. Unable to direct to details page.');
```
Comment (florkbr) [inline src/components/WorkspaceAdd/WorkspaceAddButton.tsx:71]: @kdoberst are we guaranteed that the workspace was successfully added in this case? I think it might be cleared to just throw an error indicating we were unable to redirect/navigate.

File: package.json (hunk 6 above)
Comment (florkbr) [reply package.json:52]: I think react-redux is a peer dependency of the SDK - though I think that is enforced on hac-core, not hac-infra

File: src/components/WorkspaceAdd/WorkspaceAddButton.test.tsx [hunk 27]
```diff
@@ -286,20 +297,28 @@ describe('Add Workspace modal', () => {
...
       },
     });
-    await waitFor(() => expect(screen.queryByRole('dialog')).not.toBeInTheDocument());
+
+    await waitFor(() => expect(mockedUsedNavigate).toHaveBeenCalledWith('1workspace'));
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
   });
 
-  it('Clicking cancel button closes modal', () => {
+  // test('Modal shows error if addition is successfully created but unable to verify workspace name', () => {
```
Comment (florkbr) [inline src/components/WorkspaceAdd/WorkspaceAddButton.test.tsx:305]: @kdoberst why did we disable this test?

File: src/components/WorkspaceAdd/WorkspaceAddButton.tsx (hunk 26 above)
Comment (kdoberst) [reply src/components/WorkspaceAdd/WorkspaceAddButton.tsx:71]: We are sure that the backend returned a successful status code. If it didn't then it wouldn't get to line 71 - it would immediately go to the catch block.

Line 71 is a really edge case where the api returned success but, for whatever reason, didn't pass a name back.

File: src/components/WorkspaceAdd/WorkspaceAddButton.test.tsx (hunk 27 above)
Comment (kdoberst) [reply src/components/WorkspaceAdd/WorkspaceAddButton.test.tsx:305]: I messed up.  I forgot to implement the actual test.  😕   I'll add that in the next commit

**Comment by kdoberst** [comment]
Tested that this works with the hosted version of HAC-Core (as of today) and with running HAC-Core locally using the main branch at commit [e9c4889](https://github.com/openshift/hac-core/commit/e9c48899153b826494a14b14d2010a7525b330f6)
